.github/workflows/build-snapshot.yml

```diff
@@ -9,23 +9,16 @@ jobs:
     strategy:
       matrix:
         java: [ '8' ]
-        architecture: [ 'x64' ]
         hazelcast: [ '5.1-SNAPSHOT' ]
```

Comment:
What Hz will we be compatible with? 
Ideally everything (4.2+, 4.1 is also supported but not for long..), so we should have
- latest releases of all lines here for the release build
- we should also test against latest snapshots of master, and maybe others as well

Reply:
Added here: https://github.com/hazelcast/hazelcast-hibernate/pull/341/commits/aff9ec6d5ad3b863605c37c9cbe04034e25b78ec